### PR TITLE
Support for Swift 4.2

### PR DESCRIPTION
**Pull request by jpmendel**
- Intrepid pod now has support for Swift 4.2 and will compile properly in Xcode 10.

### CODE REVIEW

File: SwiftWisdom.xcodeproj/project.pbxproj
Comment (bobgilmore) [inline SwiftWisdom.xcodeproj/project.pbxproj]: Are these new entries correct and desirable?

File: SwiftWisdom/Core/StandardLibrary/RawRepresentable/RawRepresentable.swift
Comment (bobgilmore) [inline SwiftWisdom/Core/StandardLibrary/RawRepresentable/RawRepresentable.swift]: I think that as a team, we should debate whether to deprecate *now* on a "major" release (well, as major as we get), or if we want to wait for some future point release.  I assume that it would be something along the lines of....

```swift
@available(*, deprecated, message: "Conform to CaseIterable and use allCases instead")
```

**Review by bobgilmore — COMMENTED** [review]
Looks good overall - just few minor questions.

File: SwiftWisdom.xcodeproj/project.pbxproj
Comment (jpmendel) [reply SwiftWisdom.xcodeproj/project.pbxproj]: I'm not sure how these showed up, but I'm also not sure how to make them go away aside from going in and editing the raw file.

File: SwiftWisdom/Core/StandardLibrary/RawRepresentable/RawRepresentable.swift
Comment (jpmendel) [reply SwiftWisdom/Core/StandardLibrary/RawRepresentable/RawRepresentable.swift]: I agree, it would be good to have a brief discussion about this.

**Comment by nivektric** [comment]
Hey guys - what's the status of this PR? Any chance of getting it merged in soon?

File: SwiftWisdom/Core/StandardLibrary/RawRepresentable/RawRepresentable.swift
Comment (paulrolfe) [reply SwiftWisdom/Core/StandardLibrary/RawRepresentable/RawRepresentable.swift]: Yeah, I think it's worth adding the deprecated flag as Bob mentioned. `CaseIterable` does it better than we do it.

**Review by paulrolfe — COMMENTED** [review]
Does the podspec need to update requirements for any dependencies to support 4.2. Maybe:
```ruby
rx.dependency 'RxSwift', '~> 4.2'
rx.dependency 'RxCocoa', '~> 4.2'
```
And this line too:
`s.pod_target_xcconfig = { 'SWIFT_VERSION' => '4.0' }` should probably say 4.2

**Comment by jpmendel** [comment]
I'll get to all these sometime early this afternoon and update accordingly.

**Comment by bobgilmore** [comment]
Before merging, you should do a rebase (and push --force-with-lease) to get rid of some of the redundant file changes. For example, `0de540a`, `1e3b1b5`, and `83cb616` should probably(?) all be collapsed into a single commit.

File: Intrepid.podspec
Comment (paulrolfe) [inline Intrepid.podspec]: Was I wrong about this? I can't find anything about swift 4.2 support on the RxSwift repo. Maybe `RxSwift 4.0` supports swift 4.2? 😕

Comment (jpmendel) [reply Intrepid.podspec]: It looks like `4.3.1` might, but I'm not sure. I can't find very much concrete documentation either.

Comment (paulrolfe) [reply Intrepid.podspec]: @jpmendel Found that we actually need this to be 4.3.1 according to their release notes.

**Comment by paulrolfe** [comment]
I think we should update the Jenkinsfile to specify Xcode 10.

**Comment by jpmendel** [comment]
Ok, I'll make these changes, make sure it compiles, and then @bobgilmore whenever you are finished with your App-Extension-Safe subspec PR and have it merged, I'll merge this and update the pod.

**Comment by bobgilmore** [comment]
I'm having problems re-creating the problems related to the Today Extension, so feel free to proceed without that change.  

I'd still like to see a passing Jenkins job - can you run it through Jenkins again?

**Review by bobgilmore — APPROVED** [review]

**Comment by bobgilmore** [comment]
Oh, also, as I noted at https://github.com/IntrepidPursuits/swift-wisdom/pull/155#issuecomment-426357849 , can you also do some squashing to clean up the git history a bit? Or are you planning on merging the whole thing as a "Squash and Merge?"

**Comment by jpmendel** [comment]
Yeah, was meaning to do that rebasing onto your extension commits, but if we are good to go without it, I can run through that now.

**Comment by jpmendel** [comment]
Any last calls on this before merging and updating the pod?

**Review by paulrolfe — APPROVED** [review]

**Pull request merged**
